airflow/providers/amazon/aws/triggers/rds.py

```diff
@@ -129,16 +130,21 @@ def __init__(
         waiter_max_attempts: int,
         aws_conn_id: str,
         response: dict[str, Any],
-        db_type: RdsDbType,
+        db_type: RdsDbType | str,
         region_name: str | None = None,
     ) -> None:
+        # allow passing enums for users,
+        # but we can only rely on strings because (de-)serialization doesn't support enums
+        if isinstance(db_type, RdsDbType):
+            db_type = db_type.value
+
         super().__init__(
             serialized_fields={
                 "db_identifier": db_identifier,
                 "response": response,
                 "db_type": db_type,
             },
-            waiter_name=f"db_{db_type.value}_available",
+            waiter_name=f"db_{db_type}_available",
```

Comment:
After serde this value would be `RdsDbType.INSTANCE`.
We have same problem in ECS sensors/operators in the past, see https://github.com/apache/airflow/pull/29871

Reply:
no because the enum disapears just after we enter the ctor, just above where I do
```
        if isinstance(db_type, RdsDbType):
            db_type = db_type.value
```
then we only handle strings here.

Reply:
Ahh... well ok miss that.
So then we need to validate is string value presented in `RdsDbType` otherwise

Reply:
I just want to suggest to extend 
https://github.com/apache/airflow/blob/f5c2748c3346bdebf445afd615657af8849345dd/airflow/providers/amazon/aws/utils/__init__.py#L75

To something like:

```python
class _StringCompareEnum(Enum):
    # Previous implementation instead of this comment

    def __str__(self):
        return self.value

    @classmethod
    def from_string(cls, value: str | Enum):
        if not (isinstance(value, str) or isinstance(value, cls)):
            raise TypeError(f"Unable convert {value!r} to {cls!r}.")
        elif isinstance(value, cls):
            return value

        values = cls.__members__.values()
        for v in values:
            if v == value:
                return v
        else:
            raise ValueError(
                f"Unsupported Enum value for {cls!r}. Expected one of {', '.join(repr(v.value) for v in values)}, "
                f"but got {value!r}."
            )
```

Anyway, it just nitpick it could be **entirely ignored** or we could do this later or maybe make Enums serialisable in core

Reply:
I don't think we need specific validation code, because using the wrong string will blow right away when we try to access that key in the `_waiter_arg` and `_status_paths` dictionaries.
We may want to present a more user-friendly message, but I'm not sure that's worth much added complexity, because "regular" users would never interact with this code directly anyway. I'd agree with you if this was an operator for instance.